### DISCUSSION TITLE
Dynamically update device attributes using a loop

### DISCUSSION THREAD

**Discussion by selvigp** [netbox-community/netbox#15479] [Help Wanted!]
I am facing issues while updating params in netbox devices. Have tried in two methods, one is working fine but the other one is not working. 
 
`params = {"custom_fields['CGF_OS_information']": 'Linux Red Hat', "custom_fields['CGF_Last_Discovered']": '2024-02-28 04:03', "custom_fields['CGF_CPU_Memory']": '6'}`
 
Working method:
device.custom_fields['CGF_OS_information'] = params["custom_fields['CGF_OS_information']"]
device.custom_fields['CGF_Last_Discovered'] = params["custom_fields['CGF_Last_Discovered']"]
device.custom_fields['CGF_Memory'] = params["custom_fields['CGF_Memory']"]
 
Not Working:
for param, value in params.items():
    if param == "comments":
       device.comments = value
       device.save()
   elif param.startswith("custom_fields[") and param.endswith("]"):
       field_name = param.split("[")[1].split("]")[0]
       device.custom_fields[field_name] = valueI

**Comment by markkuleinio** [comment]
Add a `print(field_name)` inside your `elif` statement and you should see that your `field_name` contains the extra single quotes. (Disclaimer: I didn't test it, just looked at it.)

Btw, adding code lines here in GitHub comments should be separated by ``` on their own lines, before and after the code snippet, to show it nicely.
